src/cli/tc.rs

```diff
@@ -91,69 +99,58 @@ pub async fn run() -> Result<()> {
     }
 }
 
-/// Checks if file or folder already exists or not.
-fn is_exists(path: &str) -> bool {
-    fs::metadata(path).is_ok()
-}
+async fn confirm_overwrite(
+    runtime: TargetRuntime,
+    file_path: impl AsRef<Path>,
+    content: &[u8],
+) -> Result<()> {
+    let file_exists = fs::metadata(file_path.as_ref()).is_ok();
 
-pub async fn init(folder_path: &str) -> Result<()> {
-    let folder_exists = is_exists(folder_path);
-
-    if !folder_exists {
+    if file_exists {
+        // confirm overwrite
         let confirm = Confirm::new(&format!(
-            "Do you want to create the folder {}?",
-            folder_path
+            "Do you want to overwrite the file {}?",
+            file_path.as_ref().display()
         ))
         .with_default(false)
         .prompt()?;
 
-        if confirm {
-            fs::create_dir_all(folder_path)?;
-        } else {
+        if !confirm {
             return Ok(());
-        };
+        }
     }
 
-    let tailcallrc = include_str!("../../generated/.tailcallrc.graphql");
-    let tailcallrc_json: &str = include_str!("../../generated/.tailcallrc.schema.json");
-
-    let file_path = Path::new(folder_path).join(FILE_NAME);
-    let json_file_path = Path::new(folder_path).join(JSON_FILE_NAME);
-    let yml_file_path = Path::new(folder_path).join(YML_FILE_NAME);
-
-    let tailcall_exists = fs::metadata(&file_path).is_ok();
+    runtime
+        .file
+        .write(&file_path.as_ref().display().to_string(), content)
+        .await?;
 
-    if tailcall_exists {
-        // confirm overwrite
-        let confirm = Confirm::new(&format!("Do you want to overwrite the file {}?", FILE_NAME))
-            .with_default(false)
-            .prompt()?;
-
-        if confirm {
-            fs::write(&file_path, tailcallrc.as_bytes())?;
-            fs::write(&json_file_path, tailcallrc_json.as_bytes())?;
-        }
-    } else {
-        fs::write(&file_path, tailcallrc.as_bytes())?;
-        fs::write(&json_file_path, tailcallrc_json.as_bytes())?;
-    }
+    Ok(())
+}
 
+async fn confirm_overwrite_yml(
+    runtime: TargetRuntime,
+    file_path: impl AsRef<Path>,
+    yml_file_path: impl AsRef<Path>,
+) -> Result<()> {
     let yml_exists = fs::metadata(&yml_file_path).is_ok();
+    let yml_path_str = yml_file_path.as_ref().display().to_string();
 
     if !yml_exists {
-        fs::write(&yml_file_path, "")?;
-
         let graphqlrc = r"|schema:
          |- './.tailcallrc.graphql'
     "
         .strip_margin();
```

Comment:
Let's create a rust type for GraphQLRC and then use `serde` to serialize it as a YML.

Reply:
done